src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/SSHUserPrivateKeyTest.java

```diff
@@ -0,0 +1,210 @@
+                CredentialsProvider.lookupStores(story.j.jenkins).iterator().next().addCredentials(Domain.global(), c);
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  withCredentials([sshUserPrivateKey(keyFileVariable: 'THEKEY', passphraseVariable: 'THEPASS', usernameVariable: 'THEUSER', credentialsId: '" + credentialsId + "')]) {\n"
+                        + "    semaphore 'basics'\n"
+                        + "    sh '''\n"
+                        + "      set +x\n"
+                        + "      echo $THEUSER:$THEPASS > out.txt\n"
+                        + "      cat $THEKEY > key.txt"
+                        + "    '''\n"
+                        + "  }\n"
+                        + "}", true));
+                WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+                SemaphoreStep.waitForStart("basics/1", b);
+            }
+        });
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.getItemByFullName("p", WorkflowJob.class);
+                assertNotNull(p);
+                WorkflowRun b = p.getBuildByNumber(1);
+                assertNotNull(b);
+                SemaphoreStep.success("basics/1", null);
+                story.j.waitForCompletion(b);
+                story.j.assertBuildStatusSuccess(b);
+                story.j.assertLogNotContains(passphrase, b);
+                FilePath out = story.j.jenkins.getWorkspaceFor(p).child("out.txt");
+                assertTrue(out.exists());
+                assertEquals(username + ":" + passphrase, out.readToString().trim());
+
+                FilePath key = story.j.jenkins.getWorkspaceFor(p).child("key.txt");
+                assertTrue(key.exists());
+                assertEquals(keyContent, key.readToString().trim());
+            }
+        });
+    }
+
+    @Test public void noUsernameOrPassphrase() throws Exception {
+        final String credentialsId = "creds";
+        final String keyContent = "the-key";
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                SSHUserPrivateKey c = new DummyPrivateKey(credentialsId, "", "", keyContent);
+                CredentialsProvider.lookupStores(story.j.jenkins).iterator().next().addCredentials(Domain.global(), c);
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  withCredentials([sshUserPrivateKey(keyFileVariable: 'THEKEY', credentialsId: '" + credentialsId + "')]) {\n"
+                        + "    semaphore 'basics'\n"
```

Comment:
Misleading semaphore name BTW; could use a more generic name (they are not shared between tests).